examples/tutorials/advanced/draping_on_3d_surface.py

```diff
@@ -0,0 +1,137 @@
+"""
+
+# %%
+
+# Load the required packages
+import pygmt
+import rasterio
+import xarray as xr
+
+# %%
+# 1. Drapping a grid
+# ------------------
+#
+# In the first example, the seafloor crustal age is plotted with color-coding on top of
+# the topographic map of an area of the Mid-Atlantic Ridge.
+
+# Define the study area in degrees East or North
+region_2d = [-50, 0, 36, 70]  # [lon_min, lon_max, lat_min, lat_max]
+
+# Download elevation and crustal age grids for the study region with a resolution of 10
+# arc-minutes and load them into xarray.DataArrays
+grd_relief = pygmt.datasets.load_earth_relief(resolution="10m", region=region_2d)
+grd_age = pygmt.datasets.load_earth_age(resolution="10m", region=region_2d)
+
+# Determine the 3-D region from the minimum and maximum values of the relief grid
+region_3d = [*region_2d, grd_relief.min().to_numpy(), grd_relief.max().to_numpy()]
+
+# %%
+# The topographic surface is created based on the grid passed to the ``grid`` parameter
+# of :meth:`pygmt.Figure.grdview`; here we use a grid of the Earth relief. To add a
+# color-coding based on *another* grid we have to pass a second grid to the
+# ``drapegrid`` parameter; here we use a grid of the crustal age. In this case the
+# colormap specified via the ``cmap`` parameter applies to the grid passed to
+# ``drapegrid``, not to ``grid``. The azimuth and elevation a the 3-D plot are set via
+# the ``perspective`` parameter.
+
+fig = pygmt.Figure()
+
+# Set up colormap for the crustal age
+pygmt.config(COLOR_NAN="lightgray")
+pygmt.makecpt(cmap="batlow", series=[0, 200, 1], reverse=True, overrule_bg=True)
+
+fig.grdview(
+    projection="M12c",  # Mercator projection with a width of 12 centimeters
+    region=region_3d,
+    grid=grd_relief,  # Use elevation grid for z values
+    drapegrid=grd_age,  # Use crustal age grid for color-coding
+    cmap=True,  # Use colormap created for the crustal age
+    surftype="i",  # Create an image plot
+    # Use an illumination from the azimuthal directions 0° (north) and 270°
+    # (west) with a normalization via a cumulative Laplace distribution for
+    # the shading
+    shading="+a0/270+ne0.6",
+    perspective=[157.5, 30],  # Azimuth and elevation for the 3-D plot
+    zsize="1.5c",
+    plane="+gdarkgray",
+    frame=True,
+)
+
+# Add colorbar for the crustal age
+fig.colorbar(frame=["x+lseafloor crustal age", "y+lMyr"], position="+n")
+
+fig.show()
+
+
+# %%
+# 2. Draping an image
+# -------------------
+#
+# In the second example, the flag of the European Union (EU) is plotted on top of a
+# topographic map of northwest Europe. This example is modified from
+# :gmt-docs:`GMT example 32 </gallery/ex32.html>`.
+# We have to consider the dimension of the image we want to drap. The image we will
+# download in this example has 1000 x 667 pixels, i.e. a aspect ratio of 3 x 2.
+
+# Define the study area in degrees East or North, with an extend of 6 degrees for
+# the longitude and 4 degrees for the latitude
```

Comment:
I just added a few lines regarding the considering the dimension of the image. Feel free to suggest improvements to the formulations.